crypto/ecdsa.go

```diff
@@ -0,0 +1,57 @@
+package crypto
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"regexp"
+)
+
+func hexToPrivateKey(priv string) (privateKey *ecdsa.PrivateKey) {
+	bytes, err := hex.DecodeString(priv)
+	if err != nil {
+		panic(err)
+	}
+
+	k := new(big.Int)
+	k.SetBytes(bytes)
+
+	curve := elliptic.P256()
+	privateKey = new(ecdsa.PrivateKey)
+	privateKey.PublicKey.Curve = curve
+	privateKey.D = k
+	privateKey.PublicKey.X, privateKey.PublicKey.X = curve.ScalarBaseMult(k.Bytes())
+	return
+}
+
+func SignEcdsa(message string, priv string) string {
+	re := regexp.MustCompile(`^[0-9a-fA-F]{64}$`)
+	if !re.MatchString(message) {
+		panic(message)
+	}
+	if !re.MatchString(priv) {
+		panic(message)
+	}
+
+	digest := sha256.Sum256([]byte(message))
+
+	privateKey := hexToPrivateKey(priv)
```

Comment:
You said that you want to know what is the purpose of the "keyFromPrivate" function used in the CodeChaini SDK JS.
https://github.com/CodeChain-io/codechain-primitives-js/blob/fa8b6d95f5576eb306257ef1184644e4db81d4b9/src/key/ecdsa.ts#L27

I checked that, and there is no cryptographical meaning. The purpose of the function is converting private key data in primitive string type to a rich type `KeyPair`.

https://github.com/indutny/elliptic/blob/475f066aebd14681591f0f0f18a2abc0ded8c390/lib/elliptic/ec/key.js#L30

Let's find why the signature result value is different after the exam.